packages/core/cache/nuxt/index.js

```diff
@@ -1,93 +1,132 @@
 /* eslint-disable */
-import path from 'path';
-import { createMiddleware } from '@vue-storefront/core/server';
+import { isAbsolute, resolve } from 'path';
 import { Logger } from '@vue-storefront/core';
 
-function rendererFactory (renderFn) {
-  const renderer = this.nuxt.renderer;
-  const renderRoute = renderer.renderRoute.bind(renderer);
+/**
+ * Helper function that imports default handler by package name
+ */
+function requirePackage (name) {
+  const path = isAbsolute(name) || name.startsWith('.')
+    ? resolve(process.cwd(), name)
+    : require.resolve(name, { paths: [ process.cwd() ] });
 
-  renderer.renderRoute = (route, context) => {
-    const render = () => renderRoute(route, context)
-    return renderFn(route, context, render)
-  }
+  return require(path).default;
 }
 
-const createInvalidators = (options) => {
-  const invalidatorsRaw = options.server.invalidators || []
+/**
+ * Adds endpoint to invalidate cache
+ */
+function createInvalidationEndpoint (driver, options) {
+  if (!options || !options.endpoint || !options.invalidators) {
+    return;
+  }
 
-  return invalidatorsRaw.map(invalidator => {
-    if (typeof invalidator === 'string') {
-      return require(path.resolve(invalidator)).default;
+  const handler = async (request, response) => {
+    try {
+      // Resolve invalidator paths to packages
+      const invalidators = options.invalidators.map(invalidator => {
+        return typeof invalidator === 'string'
+          ? requirePackage(invalidator)
+          : invalidator;
+      });
+
+      // Get tags from all invalidators
+      const tags = invalidators.reduce((tags, invalidator) => {
+        const newTags = invalidator({ request, response, options });
+        return tags.concat(newTags);
+      }, []);
```

Comment:
Maybe it would be a good idea to use `Set` instead of `Array`? We would have only unique tags

Reply:
I see we are doing it like that in other places for tags

Reply:
We use `Set` internally to collect tags on the page, but drivers and invalidators (renamed to `handlers`) should use arrays in my opinion. These functions are created by the developers, and `Set` is not so well known among junior developers compared to arrays.

However duplicate tags you mentioned might be a problem, so I added following to remove such elements:

```javascript
await driver.invalidate({
  request,
  response,
   tags: [...new Set(tags)] // Removes duplicates
});
```